spec/requests/api/actions_spec.rb

```diff
@@ -0,0 +1,54 @@
+describe "Actions API" do
+  context "Actions CRUD" do
+    let(:action) { FactoryGirl.create(:miq_action) }
+    let(:sample_action) do
+      {
+        :name        => "sample_action",
+        :description => "sample_action",
+        :action_type => "custom_automation",
+        :options     => {:ae_message => "message", :ae_request => "request", :ae_hash => {"key "=>"value"}}
+      }
+    end
+    let(:action_url) { actions_url(action.id) }
+
+    it "creates new action" do
```

Comment:
Can you add tests for unhappy paths for authorization for all these actions? And also tests for bulk queries. Thanks!